### PR TITLE
Add oAuth with Google

### PR DESCRIPTION
**Pull request by Yordanofff**
Users will be created if they don't exist
If they do, they will be 'loaded'

### CODE REVIEW

**Review by gyokhan28 — APPROVED** [review]
Reviewed and approved the changes. All secret values have been removed, and everything works as expected.

File: src/main/java/com/example/b2b_opportunities/Configs/SecurityConfig.java [hunk 1]
```diff
@@ -10,16 +10,18 @@
...
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.web.cors.CorsConfiguration;
 import org.springframework.web.cors.CorsConfigurationSource;
 import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
 
 import java.util.List;
 
+import static org.springframework.security.config.Customizer.*;
```
Comment (emiliyank) [inline src/main/java/com/example/b2b_opportunities/Configs/SecurityConfig.java:23]: We have to disable wildcard imports (with *). More info here https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

File: src/main/java/com/example/b2b_opportunities/Service/AuthenticationService.java [hunk 2]
```diff
@@ -91,4 +99,51 @@ private boolean arePasswordsMatching(UserRequestDto userRequestDto) {
...
+                .name(roleUser.name())
+                .build();
+
+        String email = (String) attributes.get("email");
+        User user = User.builder()
+                .username(email)
+                .firstName((String) attributes.get("given_name"))
+                .lastName((String) attributes.get("family_name"))
+                .email(email)
+                .provider("google")
```
Comment (emiliyank) [inline src/main/java/com/example/b2b_opportunities/Service/AuthenticationService.java:131]: It is better to extract the `google` value as a Java constant.

**Review by emiliyank — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/b2b_opportunities/Service/AuthenticationService.java (hunk 2 above)
Comment (Yordanofff) [reply src/main/java/com/example/b2b_opportunities/Service/AuthenticationService.java]: True. Fixed now.

File: src/main/java/com/example/b2b_opportunities/Configs/SecurityConfig.java (hunk 1 above)
Comment (Yordanofff) [reply src/main/java/com/example/b2b_opportunities/Configs/SecurityConfig.java]: Fixed - Applied to all files now

**Pull request merged**